### PR TITLE
Adds HLS files that previously were missed

### PR DESCRIPTION
**Pull request by sjalander**

### CODE REVIEW

File: jlm/hls/Makefile.sub [hunk 1]
```diff
@@ -47,15 +49,18 @@ libhls_HEADERS = \
...
 	jlm/hls/backend/rvsdg2rhls/GammaConversion.hpp \
 	jlm/hls/backend/rvsdg2rhls/mem-queue.hpp \
 	jlm/hls/backend/rvsdg2rhls/mem-sep.hpp \
 	jlm/hls/backend/rvsdg2rhls/memstate-conv.hpp \
 	jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.hpp \
 	jlm/hls/backend/rvsdg2rhls/remove-unused-state.hpp \
+	jlm/hls/backend/rvsdg2rhls/rhls-dne.hpp \
 	jlm/hls/backend/rvsdg2rhls/rvsdg2rhls.hpp \
 	jlm/hls/backend/rvsdg2rhls/ThetaConversion.hpp \
+	jlm/hls/backend/rvsdg2rhls/UnusedStateRemoval.hpp \
```
Comment (phate) [inline jlm/hls/Makefile.sub:63]: This file does not exist in the PR.

Comment (phate) [reply jlm/hls/Makefile.sub:63]: Okay, this file already exists in the repository, but is just not mentioned here.

**Review by phate — CHANGES_REQUESTED** [review]
Looking at the other HLS PR, I would have expected the following files in this PR:

1. dae-conv.[ch]pp
2. instrument-ref.[ch]pp
3. mem-conv.cpp. There exists no mem-conv.hpp in the other PR. However, the mem-conv.hpp is mentioned in the Makefile there.
4. rhls-dne.[ch]pp

Number 1 and 4 are here. Number 2 is missing. Any reason? Number 3 is also missing, as well as the hpp file from the other PR.

File: jlm/hls/Makefile.sub (hunk 1 above)
Comment (phate) [reply jlm/hls/Makefile.sub:63]: Ignore the above comments.

**Comment by sjalander** [comment]
They require modifications to jlm/hls/ir/hls.[hc]pp
So this PR only contains files that does not require modifications of already committed files.

**Review by phate — APPROVED** [review]

**Pull request merged**